js/framework/AnalyticsConstants.js

```diff
@@ -446,8 +449,76 @@ if (!OO.Analytics.EVENT_DATA)
   {
     var checkAdPodEndedData = OO._.bind(checkDataType, this, "AdPodEndedData");
-    this.adId = checkAdPodEndedData(adId, "adId", "string");
-  };
+    this.adId = checkAdPodEndedData(adId, "adId", ["string"]);
+  }
+
+  /**
+   * @public
+   * @class Analytics.EVENT_DATA#AdStartedData
+   * @classdesc Contains information about the type of ad that has started and its ad data.
+   * @property {string} adType The type of ad (linear video, linear overlay, nonlinear overlay)
+   * @property {object} adMetadataIn The metadata associated with the ad
+   */
+  EVENT_DATA.AdStartedData = function(adType, adMetadataIn)
+  {
+    var checkAdStartedData = OO._.bind(checkDataType, this, "AdStartedData");
+    this.adType = checkAdStartedData(adType, "adType", ["string"]);
+    this.adMetadata = checkAdType(adType, adMetadataIn);
+  }
+
+  /**
+   * @public
+   * @class Analytics.EVENT_DATA#LinearVideoData
+   * @classdesc Contains information about the linear video ad data.
+   * @property {string} adId The id of the ad
+   * @property {number} adDuration The duration of the ad video stream
+   * @property {number} adPodPosition The index of the current ad in its ad pod
+   */
+  EVENT_DATA.LinearVideoData = function(adId, adDuration, adPodPosition)
+  {
+    var checkLinearVideoData = OO._.bind(checkDataType, this, "LinearVideoData");
+    this.adId = checkLinearVideoData(adId, "adId", ["string"]);
+    this.adDuration = checkLinearVideoData(adDuration, "adDuration", ["number"]);
+    this.adPodPosition = checkLinearVideoData(adPodPosition, "adPodPosition", ["number"]);
+  }
+
+  /**
+   * @public
+   * @class Analytics.EVENT_DATA#NonLinearOverlayData
+   * @classdesc Contains information about the non linear overlay ad data.
+   * @property {string} adId The id of the ad
+   */
+  EVENT_DATA.NonLinearOverlayData = function(adId)
+  {
+    var checkNonLinearOverlayData = OO._.bind(checkDataType, this, "NonLinearOverlayData");
+    this.adId = checkNonLinearOverlayData(adId, "adId", ["string"]);
+  }
+
+  /**
+   * @public
+   * @class Analytics.EVENT_DATA#AdEndedData
+   * @classdesc Contains information about the type of ad that has ended and its ad data.
+   * @property {string} adType The type of ad (linear video, linear overlay, nonlinear overlay)
+   * @property {string} adId The id of the ad
+   */
+  EVENT_DATA.AdEndedData = function(adType, adId)
+  {
+    var checkAdEndedData = OO._.bind(checkDataType, this, "AdEndedData");
+    this.adType = checkAdEndedData(adType, "adType", ["string"]);
+    this.adId = checkAdEndedData(adId, "adId", ["string"]);
+  }
+
+  /**
+   * @public
+   * @class Analytics.EVENT_DATA#AdErrorData
+   * @classdesc Contains information about the ad error.
+   * @property {object|string} The error object or string
+   */
+  EVENT_DATA.AdErrorData = function(error)
+  {
+    var checkAdErrorData = OO._.bind(checkDataType, this, "AdErrorData");
+    this.error = checkAdErrorData(error, "error", ["string", "object"]);
```

Comment:
The multiple types is potentially used for other scenarios as well where "null" is accepted. The reason for both "string" and "object" here is because ADS_ERROR message in the AMC is reported to have either a string or object as the error parameter.